### PR TITLE
Uwp edit features

### PR DESCRIPTION
**Pull request by codergrl**
@nCastle1 please review
- Added UI for editing features and adding/editing related records in UWP. Adding features is coming in the next PR
- Moved AttributeTemplateSelector code to shared project and updated for UWP
- Improved UserPromptMessenger logic as it was failing in UWP and was not capturing user input
- Updated references to UserPromptMessenger. Now works in WPF and UWP
- Added sort for OriginRelationships
- Updated OriginRelationshipViewModel to load feature when initialized, otherwise sorting doesn't work
- Added view code in UWP to handle prompting user for input
- Updated ViewModel references to account for the refactored viewmodels

### CODE REVIEW

File: src/DataCollection.UWP/Views/FeatureEditorView.xaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                            Visibility="{Binding PopupFieldValue.ValidationError, Converter={StaticResource NullToVisibilityConverter}}" />
+                    </StackPanel>
+                </DataTemplate>
+                <DataTemplate x:Key="DateTemplate" x:Name="DateTemplate">
+                    <StackPanel>
+                        <TextBlock Style="{StaticResource BodyTextBlockStyle}" Text="{Binding PopupFieldValue.Field.Label, Mode=OneWay}" />
+                        <CalendarDatePicker
+                            Date="{Binding PopupFieldValue.Value, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged, Converter={StaticResource ObjectToDateTimeOffsetConverter}}"
+                            Loaded="CalendarDatePicker_Loaded"
+                            PlaceholderText="Select date" />
```
Comment (nCastle1) [inline src/DataCollection.UWP/Views/FeatureEditorView.xaml:44]: I recommend setting `HorizontalAlignment` to stretch, to make it consistent with the text boxes.

File: src/DataCollection.UWP/Views/FeatureEditorView.xaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                            Text="{Binding PopupFieldValue.ValidationError.Message}"
+                            TextWrapping="Wrap"
+                            Visibility="{Binding PopupFieldValue.ValidationError, Converter={StaticResource NullToVisibilityConverter}}" />
+                    </StackPanel>
+                </DataTemplate>
+                <DataTemplate x:Key="CodedValueDomainTemplate" x:Name="CodedValueDomainTemplate">
+                    <StackPanel>
+                        <TextBlock Style="{StaticResource BodyTextBlockStyle}" Text="{Binding PopupFieldValue.Field.Label, Mode=OneWay}" />
+                        <ComboBox
+                            DisplayMemberPath="Name"
```
Comment (nCastle1) [inline src/DataCollection.UWP/Views/FeatureEditorView.xaml:79]: I recommend setting `HorizontalAlignment` to stretch, to make it consistent with the text boxes.

Comment (nCastle1) [reply src/DataCollection.UWP/Views/FeatureEditorView.xaml:79]: I'd also make the change on IdentifiedFeaturePopup.xaml:L231.

File: src/DataCollection.UWP/MainPage.xaml.cs [hunk 3]
```diff
@@ -73,5 +82,64 @@ private void BladeItem_Expanded(object sender, System.EventArgs e)
             var bladeItem = sender as BladeItem;
             bladeItem.BorderThickness = new Windows.UI.Xaml.Thickness(1);
         }
+
+        /// <summary>
+        /// Event handler for displaying a message to the user
+        /// </summary>
+        private async void UserPrompt_MessageValueChanged(object sender, UserPromptMessageChangedEventArgs e)
+        {
+            var contentDialog = new ContentDialog()
```
Comment (nCastle1) [inline src/DataCollection.UWP/MainPage.xaml.cs:91]: I really like this UI for the confirmation dialogs, it feels very natural for UWP :)

**Review by nCastle1 — APPROVED** [review]
This looks really good! It feels like a really great native UWP app and I really appreciate that. I did encounter a few issues. I think they're not all related to this PR, so up to you whether you want to fix. If not, I can open an issue. 

* There's an NRE and a crash when attempting to save a feature with invalid text in a numeric field. Ideally the edit would be rejected if there are any fields with invalid values.
* I noticed that when opening the app after having previously logged in, I get an auth error. It looks like the OAuth refresh token is being set to to the username, rather than the true auth token. Culprit appears to be related to AuthViewModel.cs:L364 raising the event with the username rather than the refresh token. Changing that gets the right auth token to be stored, but there is also an issue with the credential being used to connect to portal in AuthViewModel.cs:L141.

**Comment by codergrl** [comment]
I actually noticed the OAuth issue as soon as I started working on the next branch, so it'll be fixed there, as well as the NRE exception when submitting invalid value.

**Pull request merged**